crates/accelerate/src/synthesis/linear_phase/cnot_phase_synth.rs

```diff
@@ -0,0 +1,292 @@
+                                unique_q.push(data);
+                            }
+                        }
+
+                        q = unique_q;
+
+                        for data in &mut q {
+                            let (ref mut _temp_s, _, _) = data;
+
+                            if _temp_s.is_empty() {
+                                continue;
+                            }
+
+                            for idx in 0.._temp_s.row(_j).len() {
+                                _temp_s[(_j, idx)] ^= _temp_s[(_ep, idx)];
+                            }
+                        }
+
+                        (_s, _i, _ep) = q.pop().unwrap();
+                    }
+                }
+            }
+        }
+
+        if _i.is_empty() {
+            continue;
+        }
+
+        let maxes: Vec<usize> = _s
+            .axis_iter(numpy::ndarray::Axis(0))
+            .map(|row| {
+                std::cmp::max(
+                    row.iter().filter(|&&x| x == 0).count(),
+                    row.iter().filter(|&&x| x == 1).count(),
+                )
+            })
+            .collect();
+
+        let maxes2: Vec<usize> = _i.iter().map(|&_i_idx| maxes[_i_idx]).collect();
+
+        let _temp_argmax = maxes2
+            .iter()
+            .enumerate()
+            .max_by(|(_, x), (_, y)| x.cmp(y))
+            .map(|(idx, _)| idx)
+            .unwrap();
+
+        let _j = _i[_temp_argmax];
+
+        let mut cnots0_t = vec![];
+        let mut cnots1_t = vec![];
+
+        let mut cnots0_t_shape = (0_usize, _s.column(0).len());
+        let mut cnots1_t_shape = (0_usize, 0_usize);
+        cnots1_t_shape.1 = cnots0_t_shape.1;
+        for cols in _s.columns() {
+            if cols[_j] == 0 {
+                cnots0_t_shape.0 += 1;
+                cnots0_t.append(&mut cols.to_vec());
+            } else if cols[_j] == 1 {
+                cnots1_t_shape.0 += 1;
+                cnots1_t.append(&mut cols.to_vec());
+            }
+        }
+
+        let cnots0 =
+            Array2::from_shape_vec((cnots0_t_shape.0, cnots0_t_shape.1), cnots0_t).unwrap();
+        let cnots1 =
+            Array2::from_shape_vec((cnots1_t_shape.0, cnots1_t_shape.1), cnots1_t).unwrap();
+
+        let cnots0 = cnots0.reversed_axes().to_owned();
+        let cnots1 = cnots1.reversed_axes().to_owned();
+
+        if _ep == num_qubits {
+            q.push((
+                cnots1,
+                _i.clone().into_iter().filter(|&x| x != _j).collect(),
+                _j,
+            ));
+        } else {
+            q.push((
+                cnots1,
+                _i.clone().into_iter().filter(|&x| x != _j).collect(),
+                _ep,
+            ));
+        }
+
+        q.push((
+            cnots0,
+            _i.clone().into_iter().filter(|&x| x != _j).collect(),
+            _ep,
+        ));
+    }
+
+    let state_bool = state.mapv(|x| x != 0);
+    let mut instrs = _synth_cnot_count_full_pmh(state_bool, section_size)
```

Comment:
Does this need to be mutable?

Reply:
`290 |     instructions.append(&mut instrs); ^^^cannot borrow as mutable`
because, `append` moves all data from `instrs` to `instructions` and empties `instrs`.